synthtool/gcp/common.py

```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
 
 def decamelize(str: str):
     """ parser to convert fooBar.js to Foo Bar. """
-    str2 = str[0].upper()
-    for chr in str[1:]:
-        if re.match(r"[A-Z]", chr):
-            str2 += " " + chr.upper()
+    prev = " "
+    str2 = ""
+    for i in range(0, len(str)):
+        chr = str[i]
+        if prev == " ":
+            str2 += chr.upper()
+        elif re.match(r"[A-Z]", chr):
+            # handle "JSONFrom" -> "JSON From" case.
+            next = ""
+            if i < len(str) - 1:
+                next = str[i + 1]
+            # "CoolACL" -> "Cool ACL" case.
+            if not re.match(r"[A-Z]", prev) or re.match(r"[a-z]", next):
+                str2 += " "
+            str2 += chr
         else:
             str2 += chr
+        prev = str2[len(str2) - 1]
```

Comment:
I guess this is to handle when `prev` should be `" "`? I'd prefer if the above set a temporary variable for the string to append, then this becomes

```
output += to_append
prev = to_append
```